fhir_zulip_nlp/fhir_zulip_nlp.py

```diff
@@ -113,42 +155,80 @@ def format_df(df: pd.DataFrame) -> pd.DataFrame:
     return df
 
 
-'''
-def time_format(seconds):
-    hours = seconds // 3600
-    minutes = seconds % 3600 // 60
-    secs = seconds % 3600 % 60
-    return '{:02d}:{:02d}:{:02d}'.format(hours, minutes, secs)
-'''
+def _handle_keyword_edge_cases(keyword: str, messages: List[Dict]):
+    """Filter out / etc special edge cases in API responses"""
+    new_list = []
+    for x in messages:
+        # - Filter out: Handle case where the text of one keyword is fully contained within another keyword
+        # todo: minor: it is theoretically possible that a message could contain both (i) case where the
+        #  text ofone keyword is fully contained within another keyword, as well as (ii) the actual keyword
+        #  itself, e.g. it might contain 'C-CDA' and also ' CDA ' or ' CDA,' or ' CDA.' or some other
+        #  variation. However this is really a minor edge case, and especially in the case of 'CDA' vs
+        #  'C-CDA', they are very different things, so it is unlikely that they would both appear at the
+        #  same time.
+        if keyword == 'CDA' and 'C-CDA' in x['content']:
+            continue  # don't include
+        else:
+            new_list.append(x)
+    return new_list
+
+
+# todo: this would be a good place to use `nltk`
+def _get_messages_with_context(df: pd.DataFrame, context: str) -> pd.DataFrame:
```

Comment:
Most of the context / disambiguation work (e.g. disambiguating if `$validate-code` was mentioned at the same time that `CodeSystem` or `ValueSet` was mentioned) is done here. Didn't take too much.